src/typeUtils.ts

```diff
@@ -92,7 +92,7 @@ export interface Match<TContext = any> extends MatchBase {
 }
 
 export interface Resolver {
-  resolveElements(match: Match): AsyncIterable<ResolvedElement[]>;
+  resolveElements(match: Match): AsyncIterable<ResolvedElement[]> | undefined;
```

Comment:
should this be undefined?

Reply:
good point - the inner yield type should be `undefined`, not the external one